### DISCUSSION TITLE
Asus Zephyrus G14 Temperature Limit Issue

### DISCUSSION THREAD

**Discussion by Kelk91** [seerge/g-helper#666] [Q&A]
Hello, and thank you for this fantastic tool. Unfortunately, when I attempt to apply a Temperature Limit under the slider - it does not apply permanently. I have to repeatedly use the limit between applications or upon restarting the laptop. Can I override this issue? Thank you for your time and input. Cheers.

**Comment by seerge** [comment]
Did you select auto apply checkbox? it will set that every time app starts, or laptop wakes up, etc

**Reply by Kelk91** [reply]
Hello again. I noticed that the issue persists. Even though I applied the thermal limit and checked the auto-apply box, my thermals remain inconsistent. For example, I set my thermal limit to 85C and followed your instructions, but the temps rose to 93C. However, if I manually apply the setting again, it drops down to 85C - staying there until I launch a different application or restart. Any insights?

**Reply by seerge** [reply]
Did you notice how frequently it seem to reset? i can add apply by some timer, but need to know if it should be like 1 minute or 5 seconds:)

**Reply by Kelk91** [reply]
When I use the slider to select 85C and apply, I'll run a game to test it out, and it stays at the set temp. However, when I launch a different app or restart, it returns to its factory default. Whether I use the auto-apply or not. The slider will still show as if it is set at 85C but does not reflect in its temperatures.

**Reply by seerge** [reply]
My question was about how often re-apply should happen :)

**Reply by Kelk91** [reply]
Ah, my apologies; thank you for your patience. I would imagine as frequently as possible. Five seconds? That's my best guess.

**Reply by secorlade** [reply]
@Kelk91 Hello! I'm _guessing_ that there's another app who's messing with the settings. I strongly suggest to see if this is the case, otherwise the 5 seconds reapply would conflict between the 2 apps. 

For applying settings on restart, try adding a 10 second delay to the GHelper task in "Task Scheduler".
I did this because GHelper loaded inconsistently for me after restarting, where 4/10 restarts the app icon was missing from the tray and settings weren't even applied. After adding the delay and testing with ~20 restarts I had no more issues.

@seerge Hello! I'm not against a "reapply every 5 seconds functionality", I'm voting pro for it. It would make things a lot more smoother in the long run.

**Reply by seerge** [reply]
@Kelk91 @secorlade I'm inviting you to test a built that would re-apply temp limit (only temp, not undervolting) every 5 seconds : 

Re-apply things will work only if "auto apply" is checked :) 

[GHelper.zip](https://github.com/seerge/g-helper/files/11881737/GHelper.zip)

**Reply by secorlade** [reply]
@seerge if you make it for undervolting as well, let me know. My laptop can handle -30 with no problems.

**Reply by seerge** [reply]
@secorlade but do you encounter UV resetting by itself ? cause if temp limit indeed seem to (randomly) reset sometimes. UV seem to be super stable.

I just don't want to overload app with constant re-applying everything possible, as i don't know if won't cause more troubles ...

**Reply by secorlade** [reply]
@seerge It's a sensible topic. The issue with temps indeed was reproduceable (even randomly during a 30 minute test with Cinebench R23). But with undervolting I'm not entirely sure. I'll attempt a series of tests in the weekend since I need to monitor HWInfo reports closely and compare them between runs with -30 and 0.

Atm I started a 30 minute multicore stress test with 0.92.0.0 and after 5 minutes temps seem fine, averaging at 85.5 C with a 85 C limit. Will run a few of these tests and report later.

**Reply by secorlade** [reply]
@seerge After 2 x 30 minutes Cinebench tests I can report everything seems fine. Will continue to monitor and report back anything weird.

Settings: 
CPU Temp limit: 85 C
Undervolt CPU: -30
Undervolt iGPU: -20

I attached HWInfo logs for 1 of these tests
[30-minute-test.xlsx](https://github.com/seerge/g-helper/files/11882567/30-minute-test.xlsx)

**Reply by Kelk91** [reply]
@seerge That has done the trick! After running a few games and torture testing with Prime95 - the temperature sticks to the set 85C. I noticed some stuttering in-game, but I wonder if it's related to temperature limit and performance throttling.

**Reply by Kelk91** [reply]
@secorlade You have a great point, but I have triple-checked... quadruple-checked at the time and could not find a conflict. I'm running a barebones system with no startup applications besides GHelper. I also completely removed all ASUS-related software. The delay is not needed because GHelper runs upon login with no issue. Random question for you, though - what BIOS are you running?

**Reply by secorlade** [reply]
@Kelk91 I have an Asus Zephyrus Duo 16 (2023) running on bios version 315. I know it's not the same model with yours, but the issue was reproducing for me as well.

**Reply by seerge** [reply]
@Kelk91 i think stuttering was due to throttling. CPU in my G14 due to vapor chamber heats very fast in games, even when not loaded (it heats from GPU), so when I have set temp limit to 80C power had to drop to something like 5-10W ...

You can try this build, I have set re-apply limit to 30 seconds. I think this should be good enough for an average user and at the same time not too often (and on this stage it seems like re-apply for temps is needed anyway)

[GHelper.zip](https://github.com/seerge/g-helper/files/11882887/GHelper.zip)

**Reply by Kelk91** [reply]
@seerge What are your current settings for the temp limit and voltage so I can replicate it on my application?

**Reply by seerge** [reply]
@Kelk91 my point was that setting a limit to low will inevitably lead to strong throttling while gaming
